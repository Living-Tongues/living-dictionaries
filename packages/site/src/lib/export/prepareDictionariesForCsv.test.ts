import type { IDictionary } from '@living-dictionaries/types';
import {
  prepareDictionariesForCsv,
  timestamp_to_string_date,
  create_dictionary_url,
} from './prepareDictionariesForCsv';

describe('create_dictionary_url', () => {
  test('returns a dictionary URL', () => {
    const dictionary_id = 'foo';
    expect(create_dictionary_url(dictionary_id)).toEqual('https://livingdictionaries.app/foo');
  });
});

describe('timestamp_to_string_date', () => {
  test('converts createdAt to readable string', () => {
    const timpestamp = {
      seconds: 1591088635,
      nanoseconds: 880000000,
    };
    expect(timestamp_to_string_date(timpestamp)).toEqual('Tue Jun 02 2020');
  });
  test('returns an empty string if timestime is falsy', () => {
    const timestamp = undefined;
    expect(timestamp_to_string_date(timestamp)).toEqual('');
  });
});

describe('prepareDictionariesForCsv', () => {
  const dictionaries: IDictionary[] = [
    {
      glottocode: 'badh1238',
      glossLanguages: ['en', 'hi', 'pa'],
      createdAt: {
        seconds: 1669080253,
        nanoseconds: 340000000,
      },
      createdBy: 'OeVwHacvXMTW0ocKuidCWxBrCIP2',
      updatedBy: 'OeVwHacvXMTW0ocKuidCWxBrCIP2',
      name: '<xaxdeleted>',
      entryCount: 0,
      public: false,
      alternateNames: [],
      coordinates: {
        latitude: 30.133,
        longitude: 79.539,
      },
      iso6393: '',
      id: 'badhani',
    },
    {
      createdBy: 'qoP1VCYTaEWb9i3CEH16TcpoUNI3',
      updatedBy: 'T4qikh1eTafizvpmHNcG29uRQ2j1',
      coordinates: {
        latitude: 51.833,
        longitude: 107.616,
      },
      glossLanguages: ['en', 'ru', 'mn', 'cmn'],
      public: false,
      entryCount: 1,
      iso6393: 'bua',
      glottocode: 'buri1258',
      alternateNames: ['Buriat', 'Buriad', 'Bargu'],
      name: 'Buryat',
      id: 'buryat',
    },
    {
      alternateNames: null,
      updatedBy: 'FtP0LHfC9SMzyfcWN48HaNclSxC3',
      glossLanguages: ['ig'],
      name: ' IGBO LANGUAGE (ASUSU IGBO)',
      public: false,
      location: 'South Eastern States of Nigeria, West Africa',
      createdBy: 'vLfYz1Fja8SlL5dSIV7spGeFCXj1',
      entryCount: 1,
      id: 'igbo-language-(asusu-igb',
    },
  ];
  test('admin/dictionaries example', () => {
    const admin = 2;
    expect(prepareDictionariesForCsv(dictionaries, admin)).toMatchSnapshot();
  });
  test('List of dictionaries example', () => {
    const admin = 0;
    expect(prepareDictionariesForCsv(dictionaries, admin)).toMatchSnapshot();
  });
});
