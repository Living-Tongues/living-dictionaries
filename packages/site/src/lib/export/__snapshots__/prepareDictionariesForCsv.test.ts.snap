// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prepareDictionariesForCsv > List of dictionaries example 1`] = `
[
  {
    "glottocode": "Glottocode",
    "iso6393": "ISO 639-3",
    "latitude": "Latitude",
    "location": "Location",
    "longitude": "Longitude",
    "name": "Dictionary Name",
    "public": "Public",
    "thumbnail": "Thumbnail",
    "url": "URL",
  },
  {
    "glottocode": "badh1238",
    "iso6393": "",
    "latitude": "30.133",
    "location": "",
    "longitude": "79.539",
    "name": "<xaxdeleted>",
    "public": "false",
    "thumbnail": undefined,
    "url": undefined,
  },
  {
    "glottocode": "buri1258",
    "iso6393": "bua",
    "latitude": "51.833",
    "location": "",
    "longitude": "107.616",
    "name": "Buryat",
    "public": "false",
    "thumbnail": undefined,
    "url": undefined,
  },
  {
    "glottocode": undefined,
    "iso6393": undefined,
    "latitude": undefined,
    "location": "South Eastern States of Nigeria_ West Africa",
    "longitude": undefined,
    "name": " IGBO LANGUAGE (ASUSU IGBO)",
    "public": "false",
    "thumbnail": undefined,
    "url": undefined,
  },
]
`;

exports[`prepareDictionariesForCsv > admin/dictionaries example 1`] = `
[
  {
    "alternate_names": "Alternate Names",
    "alternate_orthographies": "Alternate Orthographies",
    "author_connection": "Author Connection",
    "community_permission": "Community Permission",
    "conlang_description": "Conlang Description",
    "created_at": "Created At",
    "entries": "No. Entries",
    "gloss_languages": "Gloss Languages",
    "glottocode": "Glottocode",
    "iso6393": "ISO 639-3",
    "language_used_by_community": "Language Used By Community",
    "latitude": "Latitude",
    "location": "Location",
    "longitude": "Longitude",
    "name": "Dictionary Name",
    "public": "Public",
    "thumbnail": "Thumbnail",
    "url": "URL",
    "video_access": "Video Access",
  },
  {
    "alternate_names": "",
    "alternate_orthographies": undefined,
    "author_connection": undefined,
    "community_permission": undefined,
    "conlang_description": undefined,
    "created_at": "Mon Nov 21 2022",
    "entries": "0",
    "gloss_languages": "en, hi, pa",
    "glottocode": "badh1238",
    "iso6393": "",
    "language_used_by_community": undefined,
    "latitude": "30.133",
    "location": "",
    "longitude": "79.539",
    "name": "<xaxdeleted>",
    "public": "false",
    "thumbnail": undefined,
    "url": undefined,
    "video_access": undefined,
  },
  {
    "alternate_names": "Buriat, Buriad, Bargu",
    "alternate_orthographies": undefined,
    "author_connection": undefined,
    "community_permission": undefined,
    "conlang_description": undefined,
    "created_at": "",
    "entries": "1",
    "gloss_languages": "en, ru, mn, cmn",
    "glottocode": "buri1258",
    "iso6393": "bua",
    "language_used_by_community": undefined,
    "latitude": "51.833",
    "location": "",
    "longitude": "107.616",
    "name": "Buryat",
    "public": "false",
    "thumbnail": undefined,
    "url": undefined,
    "video_access": undefined,
  },
  {
    "alternate_names": undefined,
    "alternate_orthographies": undefined,
    "author_connection": undefined,
    "community_permission": undefined,
    "conlang_description": undefined,
    "created_at": "",
    "entries": "1",
    "gloss_languages": "ig",
    "glottocode": undefined,
    "iso6393": undefined,
    "language_used_by_community": undefined,
    "latitude": undefined,
    "location": "South Eastern States of Nigeria_ West Africa",
    "longitude": undefined,
    "name": " IGBO LANGUAGE (ASUSU IGBO)",
    "public": "false",
    "thumbnail": undefined,
    "url": undefined,
    "video_access": undefined,
  },
]
`;
